src/features/session_replay/aggregate/index.js

```diff
@@ -112,23 +102,34 @@ export class Aggregate extends AggregateBase {
       }
     }, this.featureName, this.ee)
 
+    // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
+    // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
+    registerHandler('sr-errorAgg', (e) => {
+      this.errorNoticed = true
+      if (this.recorder) this.recorder.currentBufferTarget.hasError = true
+      // run once
+      if (this.mode === MODE.ERROR && globalScope?.document.visibilityState === 'visible') {
+        this.switchToFull()
+      }
+    }, this.featureName, this.ee)
+
     const { error_sampling_rate, sampling_rate, autoStart, block_selector, mask_text_selector, mask_all_inputs, inline_stylesheet, inline_images, collect_fonts } = getConfigurationValue(this.agentIdentifier, 'session_replay')
 
-    this.waitForFlags(['sr']).then(([flagOn]) => {
-      this.entitled = flagOn
+    this.waitForFlags(['srs', 'sr']).then(([srMode, entitled]) => {
+      // waitForFlags now returns the raw value (0, 1) of the flag.
+      // Eventually BCS will report this as (0, 1, 2) for the appropriate mode, and report entitlements separately.
+      // This will need to be fixed to honor that when that change is made. Returning the raw flag value enables that work.
+      this.entitled = !!entitled
```

Comment:
Just saying
```suggestion
      this.entitled = Boolean(entitled)
```

Reply:
webpack transpiles that to `!!` anyway since its shorter, just fyi